src/main/java/de/fraunhofer/aisec/cpg/graph/declarations/FunctionDeclaration.java

```diff
@@ -100,6 +101,12 @@ public String getSignature() {
         + Objects.requireNonNullElse(this.type, UnknownType.getUnknownType()).getTypeName();
   }
 
+  public boolean hasSameSignature(FunctionDeclaration targetFunctionDeclaration) {
+    return targetFunctionDeclaration.getName().equals(this.name)
+        && targetFunctionDeclaration.getType().equals(this.type)
```

Comment:
The return-type is normally not part of the function signature that influences call resolution. At least not in C++ or Java. This may lead to issues if we use it for call resolution. Is the check for type necessary for some other check?